app/src/policy_engine.h

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "cose.h"
+#include "http_error.h"
+#include "profiles.h"
+#include "tracing.h"
+
+#include <ccf/js/common_context.h>
+#include <string>
+
+namespace scitt
+{
+  using PolicyScript = std::string;
+
+  namespace js
+  {
+    static inline ccf::js::core::JSWrappedValue claim_profile_to_js_val(
+      ccf::js::core::Context& ctx, ClaimProfile claim_profile)
+    {
+      switch (claim_profile)
+      {
+        case ClaimProfile::IETF:
+        {
+          return ctx.new_string("IETF");
+          break;
```

Comment:
return + break?..

Reply:
Sure! It's a bit of unreachable code that a cautious linter might complain about, but I think it's a sensible pattern - if you replace the `return`s with some local assignment, you're reminded that you're inside a switch statement.

Reply:
It just seems too much to me, such as `if (p == nullptr)` instead of `if (!p`, or `5 == x` instead of `x == 5` to prevent accidental assignment.

For reference: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions.

But I'm fine to leave it here, just saying.

Reply:
@eddyashton we could -Wimplicit-fallthrough if we're worried about this, since we have -Werror?

Reply:
There's a few instances in the code base of returns inside switch statements without a following break (`get_jose_alg_from_cose_alg`, `OperationsIndexingStrategy::lookup`), so I've removed the `break` to match local style.

FWIW, I don't like the core guideline's brevity here - I think `if (p)` for pointers leads to `if (x)` for lots of other expressions with implicit bool conversions, leading to missed errors when the type changes (or for odd implementations of bool operator). eg, if I change a `T x` to `std::optional<T> x`, I'd rather get a compile-error on every existing `if (x.is_good())` (changing to `if (x.has_value())` or `if (x->is_good())` as appropriate), rather than having a bunch of `if (x)` that continue to compile but mean something different.

Anyway this is a very pretty bike shed we've built.

Reply:
Rust...